### ISSUE TITLE
HMR Infinite Loop with wss settings

### ISSUE THREAD

**Issue by stolinski** [frandiox/vite-ssr#24]
Our vite config has the following config to connect via ssl in dev.
```
  server: {
    hmr: {
      host: 'leveluptutorials.dev',
      port: 443,
      protocol: 'wss',
    },
  },
```
This config works fine with the standard vite cli, but causes an infinite loop using vite ssr. It's hard to debug because of the nature of things, but it seems like the wss connection is failing causing a refresh on loop. 

The network shows a 502 request.
```
Request URL: https://leveluptutorials.dev/__vite_ping
Request Method: GET
Status Code: 502 
Remote Address: 127.0.0.1:443
Referrer Policy: strict-origin-when-cross-origin
```

It's kind of tough to put together a reproduction given the ssl requirements, I can provide as much as possible to help and can dive into your code to see if there is anything that sticks out.

Edit:
Some additional information. The looping we're seeing is the same as if the hmr config settings were not set at all. This tells me that this library might not be seeing the server.hmr options.

**Comment by stolinski** [comment]
Small update here. I'm seeing that Vite SSR is getting the hmr settings, because it's getting the correct host and port, but for some reason it's not connecting when using SSR

**Comment by frandiox** [comment]
@stolinski Hi!
I've never tried SSL or WS in Vite so right now I don't know much about this issue.
However, I've noticed that there was a difference in how WS was setup in `vite-ssr` vs `vite`. Perhaps is the source of the issue, but maybe not.

I've created a [PR](https://github.com/frandiox/vite-ssr/pull/25) refactoring the dev-server and the usage of WS should now be similar to Vite.
If you have time, checkout that branch and try it with your SSL/WS configuration.
